### ISSUE TITLE
Parameters not rendered correctly in 2.7.0 UI

### ISSUE THREAD

**Issue by matteoannotell** [apache/airflow#33923]
### Apache Airflow version

2.7.0

### What happened

When initialising integer parameters as 0, or arrays as empty list, the default behaviour of the UI is to render it as null.
This results in default parameters being ignored unless re-imputed manually.
Passing a config like this:
````
params={
        "integer_param": Param(0, type="integer", minimum=0),
        "list_param": Param([], type="array"),
    },
````
results in this in the UI when the DAG loads, forcing the user to manually input again 0, or editing the json for having an empty list
<img width="1699" alt="image" src="https://github.com/apache/airflow/assets/111278183/e0722be8-1806-4cf0-85b3-ec422c8f96a7">

<img width="371" alt="image" src="https://github.com/apache/airflow/assets/111278183/f9670f83-5d29-48f5-a8ad-7e4432c1ba47">

### What you think should happen instead

0 rendered as 0, empty lists as empty lists

### How to reproduce

This DAG should be an MVP for reproducing the error

````
from datetime import datetime

from airflow import DAG
from airflow.models.param import Param
from airflow.operators.python_operator import PythonOperator

default_args = {
    "owner": "data-enablement",
    "start_date": datetime(2023, 4, 24),
    "retries": 30,
    "startup_timeout_seconds": 600,
}

with DAG(
    "test-error",
    default_args=default_args,
    schedule=None,
    params={
        "integer_param": Param(0, type="integer", minimum=0),
        "list_param": Param([], type="array"),
    },
    render_template_as_native_obj=True,
    description="MVP for reproducing error",
) as dag:
    print_int_task = (
        PythonOperator(
            task_id="print_int",
            op_args=[
                "{{ params.integer_param}}",
            ],
            python_callable=(lambda integer_param: print(integer_param)),
        ),
    )

    print_list_task = PythonOperator(
        task_id="print_list",
        op_args=[
            "{{ params.list_param}}",
        ],
        python_callable=(lambda list_param: print(list_param)),
    )

print_int_task, print_list_task
`````

### Operating System

macOS M1 locally

### Deployment

Docker-Compose

### Deployment details

Using this image:
https://hub.docker.com/layers/apache/airflow/2.7.0-python3.10/images/sha256-d6cb37572c21546ba3e60274d3cc1e782cc517a31443b8051702b19584e7379b?context=explore

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by SamWheating** [comment]
I can look into this, feel free to assign to me.

**Comment by josh-fell** [comment]
All yours @SamWheating!

**Comment by jens-scheffler-bosch** [comment]
Empty lists being rendered as `null` (also the same for empty strings and objects) was actually a desire. This is the (only?) way to enforce users to enter values into lists for the required input check.
If you want to have lists optional then you need to use `type=["null", "array"]`.

I agree the integer thing is actually a bug. With `0` as default value a `0` should be accepted.

**Comment by matteoannotell** [comment]
> Empty lists being rendered as null (also the same for empty strings and objects) was actually a desire.

But then if one defaults the value of a parameters to an empty list, and build everything that uses it around the premises of receiving an iterable, things will break when receiving `None`, which is not iterable?

**Comment by SamWheating** [comment]
I agree that there should be a way to pass either a null or an empty list through the form interface without any workarounds, but I think it will require some more changes (maybe we'll have to include the `[ ]` in the form field?).

I'll draft up some possible implementations and we can chat on the PR.

**Comment by jens-scheffler-bosch** [comment]
If the field is a list and is required anyway then I would expect that also at least one element needs to be returned - an empty list for me does not represent a valid value if defined as `Param([], type="array")`. Or is your use case expecting that the user can pass no element in the array as valid option?

The main difference we have is how to define a "non required or empty array" as result and technically all is possible. But the form can not distinguish whether `[]` or `None` is expected.
We have the same matter also with all other non-bool types (string, int, object), an empty string field also is passed as `None`. I still opt for consistency.

**Comment by SamWheating** [comment]
> We have the same matter also with all other non-bool types (string, int, object), an empty string field also is passed as None. I still opt for consistency.

I think that for any of these there is a clear difference between "empty" values (`""`, `0`, `{}`, `[]`) and null values, and we should _probably_ allow users to provide either the empty or null value as required. So while I agree that we should have consistency, I think I disagree on what that consistent interface should look like.

I also acknowledge that this definitely adds some complexity implementation-side and probably won't make any difference for 99% of use cases.

> Or is your use case expecting that the user can pass no element in the array as valid option?

Yes, I think that both `null` and `[]` are valid but distinctly different.

**Comment by matteoannotell** [comment]
> Or is your use case expecting that the user can pass no element in the array as valid option?

Yes, empty list would be a valid option in my DAG (with all discussed so far still holding, I could just rewrite the implementation so that null is allowed and the rest of the DAG is adjusted accordingly)

**Comment by jens-scheffler-bosch** [comment]
I'd propose to close this ticket and separate-out the empty list thing into a second discussion/feature ticket. This will not make it (including discussion) into 2.7.1.
The main bug of missing rendering a `0` number value was fixed in #33965
